### PR TITLE
[BEAM-1371] Reduce cost of sample combine test

### PR DESCRIPTION
**Pull request by robertwb**
There's no need to run this test 300 times, which takes tens of seconds.

Also split global and per-key sampling into separate tests.

Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [ ] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

### CODE REVIEW

**Comment by robertwb** [comment]
R: @chamikaramj

File: sdks/python/apache_beam/transforms/combiners_test.py [hunk 1]
```diff
@@ -240,7 +238,7 @@ def match(actual):
       assert_that(result, matcher())
       pipeline.run()
```
Comment (dhalperi) [inline sdks/python/apache_beam/transforms/combiners_test.py]: Dumb Q: is the pipeline.run() supposed to be outside the loop?

Comment (robertwb) [inline sdks/python/apache_beam/transforms/combiners_test.py]: The pipeline creation is in the loop, so the run is as well. However, I'll change it to run a single pipeline with multiple samples instead.

**Comment by chamikaramj** [comment]
Seems like there are some lint errors.

**Comment by chamikaramj** [comment]
LGTM

**Comment by dhalperi** [comment]
retest this please

(maven was down earlier)

**Comment by dhalperi** [comment]
please tag with a JIRA before merging

**Comment by robertwb** [comment]
What's the threshold for a JIRA (which are fairly heavyweight)? This is basically "I cleaned up a test I noticed while working on another ticket" (though it did evolve from the original 1-character change) and I don't think that meets the bar.

**Comment by dhalperi** [comment]
The current threshold is "all changes need a JIRA" (note the github template makes this clear).

There is definitely a need for a `[codehealth]` tag or a `[trivial]` tag. We need to start this discussion.

But in general, I'd say if it has a nonzero change of causing flakes (e.g., affects any test code, pipeline structure, etc.) it needs some public justification that is on an Apache-managed property. We've had build breaks because someone upgraded a dependency with no explanation, and it's hard to know whether it's safe to rollback without understanding why the change was made.

This is especially crucial given how global Beam is -- people are off time zone and don't have everyone reachable instantaneously via chat.

**Comment by robertwb** [comment]
I'd say that requiring a JIRA issue for cleanups discourage code health (or, possibly, worse, encourage always folding such changes into other, unrelated PRs). For this PR I created and closed a ticket. I'll start a discussion on what to do in general. 

Of course justification for upgrading a dependency, which is a bigger change, should be in the commit itself... (commit messages should say why, not just what, as what can often be inferred from the code).

**Pull request closed without merging**